### PR TITLE
Fix map obj syntax for Firefox.

### PR DESCRIPTION
**Pull request by jimafisk**
Maps were not rendering in FireFox due to this error message: `TypeError: "map--144161" is read-only` as noted on issue https://github.com/CityOfBoston/boston.gov/issues/957

Details about this error can be found [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Read-only?utm_source=mozilla&utm_medium=firefox-console-errors&utm_campaign=default).

Fixed the object syntax (was using PHP style array) so this functions properly in strict mode.

### CODE REVIEW

File: scripts/components/map.js [hunk 1]
```diff
@@ -64,18 +64,18 @@ var BostonMap = (function () {
     var zoom = mapObj.componentZoom ? mapObj.componentZoom : mapObj.esriZoom ? mapObj.esriZoom : 14;
 
     // Apply default coordinates and zoom level.
-    map[map[0].id] = L.map(mapID, {zoomControl: false}).setView([latitude, longitude], zoom);
+    map.mapID = L.map(mapID, {zoomControl: false}).setView([latitude, longitude], zoom);
```
Comment (finneganh) [inline scripts/components/map.js]: Gotta clean this up a bit more. `map[0].id` is used above still. Also, this is setting the value of a `mapID` property to a Leaflet Map object, which will be confusing down the road.

Comment (finneganh) [reply scripts/components/map.js]: LMK if you want to chat / debug in person.

Comment (jimafisk) [reply scripts/components/map.js]: Thanks for the help on this @finneganh! I updated the variable names so it's more clear what they represent, which simplifies the conditional logic. 

I also only declared the map ID once even though this value can be found in the object that Drupal passes via `mapData` or in the page template.

Comment (finneganh) [reply scripts/components/map.js]: @jimafisk Don't forget to change these bits to just `var map = L.map(…)`

Comment (jimafisk) [reply scripts/components/map.js]: Ah yes, I see how that's unnecessary now. Should be all set. Thanks for pointing that out!

File: scripts/components/map.js [hunk 2]
```diff
@@ -64,18 +64,18 @@ var BostonMap = (function () {
     var zoom = mapObj.componentZoom ? mapObj.componentZoom : mapObj.esriZoom ? mapObj.esriZoom : 14;
 
     // Apply default coordinates and zoom level.
-    map[map[0].id] = L.map(mapID, {zoomControl: false}).setView([latitude, longitude], zoom);
+    map = L.map(mapID, {zoomControl: false}).setView([latitude, longitude], zoom);
```
Comment (finneganh) [inline scripts/components/map.js]: `var map`

**Comment by finneganh** [comment]
LGTM after you declare the `map` with `var` and squash the commits.

**Pull request merged**